### DISCUSSION TITLE
(Qual | Que tipo De) desenvolvedor você quer ser daqui 2 anos?

### DISCUSSION THREAD

**Discussion by felquis** [frontendbr/forum#407] [Discussão]
Vocês pensam em algo do tipo?

**Comment by suissa** [comment]
Do tipo que esteja ensinando programação para qualquer um. <3

**Comment by woliveiras** [comment]
Eu quero ser do tipo que resolve problemas da melhor maneira!

Fora isso quero, assim como o @suissa, transmitir muito conhecimento pra galera.

Pra isso eu tenho estudado bastante, tenho uma espécie de trilha de conhecimentos do que quero ter daqui a algum tempo e (por coincidência) tenho meus planos já traçados para daqui a 2 anos. - Na realidade eu tinha 3 anos planejados, mas ano passado foi o primeiro desses três. 😄

**Comment by paulinhapenedo** [comment]
Assim como o @woliveiras, eu quero do tipo que resolve problemas da melhor maneira. Mesmo que isso signifique não usar o framework/lib da moda.

Desde que o usuário final da minha aplicação esteja feliz e os desenvolvedores entendam e consigam dar manutenção no código que escrevi, eu cumpri minha missão.

**Comment by joaoeffting** [comment]
Coloquei como meta para daqui 2/3 anos eu ter um conhecimento sólido sobre JavaScript (Vanilla). No meio desse tempo também vou me aprofundar mais no React e todo o core que rodeia ele (não quero apenas saber utilizá-los, mas quero aprender também o "under the hood" de cada um)

Além disso, como já citaram, quero poder ficar mais proficiente com meu trabalho e ser referência pros meus colegas de trabalho e de comunidade.

E claro que, secretamente, trabalhar no plano para que o JavaScript domine o mundo :D

**Comment by rubenmarcus** [comment]
- Trabalhar remoto, com possibilidade de dar um boa qualidade de vida pra mim e família
- conseguir trabalhar viajando
- fazer projetos pessoais que eu me orgulhe ( jogos, animação , 3d ),não apenas viver para projetos comerciais ou clientes
- contribuir para a comunidade
- trabalhar menos horas e ganhar mais

Pra isso:
- estudar,estudar,estudar
está nos planos tudo que o pessoal falou ai (maturidade em JS e em código em geral) e ter um ritmo e dinamica mais organizado..

Meu foco mesmo é não depender de empresa X ou freela Y pra ter liberdade e independência. No final o que eu quero mesmo é ter tempo pra minha família e meus hobbies, e ganhar o suficiente para ter uma vida modesta mas com qualidade, suficiente para salvar alguma grana pro futuro também.


Hoje eu basicamente trabalho para fechar as contas do mês. Se quero investir em algo que gosto, ou tenho uma emergência, preciso recorrer ao crédito. Quero sair desse loop.

**Comment by thulioph** [comment]
To no mesmo ritmo do @rubenmarcus, principalmente:

- trabalhar menos horas e ganhar mais
- trabalhar viajando

**Comment by rubenmarcus** [comment]
Uma coisa que eu não falei, mas esse ano comecei fazendo um trabalho voluntário com uma ong a custo zero,pra eles. Isso é algo que eu quero manter. Pelo menos um projeto por ano,pra alguém que não poderia pagar pelo meu trabalho, e que sei que vai fazer um impacto social na vida de alguém e a diferença para uma instituição...

**Comment by malaquiasdev** [comment]
@rubenmarcus opa uma iniciativa diferente, poderia falar mais sobre isso?

**Comment by JoseCage** [comment]
Bem interessante essa sua iniciativa @rubenmarcus..

Deu uma ideia até :+1:

**Comment by leonardoelias** [comment]
@rubenmarcus  @mmalaquias1  @JoseCage 

Eu tbm tenho alguns projetos assim. 
Eu lembro que comecei a fazer, quando eu vi isso aqui : http://adoteumbriefing.org/

Se vocês tiverem interesse, entro nesse projeto ae tbm :) haha

**Comment by rubenmarcus** [comment]
Opa pessoal, no começo do ano eu decidi que eu ia fazer algo diferente esse ano, e disponibilizar um pouco do meu tempo,e conhecimento (mesmo que não tenha muito dos dois) Em transmitir o pouco que eu tenho para os outros de forma gratuita. 

Bom, para isso, eu entrei no site Atados.com.br , e minha idéia era trabalhar com crianças em escola pública ou refugiados, ensinando o básico sobre desenvolvimento web.

Acontece que na mesma época eu comprei uma Canon T5, e fotografia é um dos meus hobbies,que eu não levava tão a sério, e comecei a levar mais após ser demitido da Centauro, e meu chefe falar que gostava das minhas fotos e que eu tinha que começar a levar mais a sério , "esse hobby"

Então me vi de frente de diversos projetos voluntários aqui de SP, e me inscrevi em uns 5 de web ,e em um de fotografia (uma ONG da Zona Leste - 2h da onde eu moro)

Aconteceu que os de web ninguém me chamou, e muitos eu via também que eram ongs e empresas com recursos que queriam sites de graça, e outros eram projetos sérios que já tinham muitos candidatos ou apoio público e privado

Pois bem, essa ONG teve 80 candidatos a fotografos, mas segundo a administradora, nenhum respondeu, eu fui o único, Então combinamos e eu fui na sede em Março e sentamos, e ela queria que fosse feito um documentário a respeito da vida dos catadores de materiais recicláveis da ZL, e eu topei claro, porque eu moro em bairro de periferia e tento manter pelo menos um pouco de consciência social e fugir um pouco do rótulo de sermos uma "sociedade de consumo"

Pois bem esse projeto ainda está em fase de transformação , e eu produzi um mini documentário que a adminstradora da ong apresentou no TCC dela, e estamos buscando parceiros públicos e privados para realizações das ações da ONG junto aos catadores, e de conscientização ambiental e social.


Ai, além de fotografia, eu sugeri também, fazer o site da ong ( hoje ele já existe, está na locaweb e foi feito com construtor de sites) , vou reformular o site da ong, e o site do projeto com catadores está feito ( é um template wordpress, que está cheio de bugs visuais, e o conteúdo eu escrevi com as minhas idéias, só essa semana a administradora da ong e outros colaboradores enviaram o material oficial )


aqui estão os links:


**Site em reformulação que só tem divulgação interna por enquanto :** http://www.maiscatador.org
**Site da Ong antigo que vou reformular até o meio de Fevereiro :** http://www.cemais.org

**Facebook do Projeto:**
https://www.facebook.com/maiscatador/

**Vídeo:**
https://www.facebook.com/maiscatador/videos/1332990276743638/



Todo esse material eu que produzi (tirando o site Cemais). Claro com o consentimento dos catadores e também da administradora e colaboradores da Ong.

Fiz o logo do projeto, identidade visual, novo logo da ong, o vídeo, as fotos, ..  É bastante coisa, e eu ainda preciso revisar o audio dos videos, preciso deixar o logo em alta ,e melhorar os sites..

Mas bom pra mim foi um passo importante, e não tem nada que te pague, ver um catador de materiais reciclaveis que ganha 800 reais num mês, todo orgulhoso e chorando porque se viu numa projeção de vídeo que alguem deu espaço para ele falar da realidade dele. 

Ou ele andar pelo bairro,com um jornal do bairro mostrando pra todo mundo a foto no jornal dele que você tirou.

Quero falar pro pessoal que eu não sou nenhum documentarista e videomaker profissional, nem fotografo, e desenvolvedor e designer estou bem longe de chegar nas pessoas que tenho como referência. 

Mas é que senti que em 2016 eu tinha que começar a fazer algo mais do que eu sabia.. E aproveitar também e juntar um hobbie que eu tinha ( tirar fotos da cidade, de viagens e de pessoas na rua que eu sempre gostei ) , e trabalhar em prol de uma causa social real.

Enfim é isso.
